### PR TITLE
PP-94: Revenue Sharing Model

### PR DESCRIPTION
**Pull request by mortelli**
# Revenue Sharing Model

This PR is part of the implementation of the Revenue Sharing model feature.

## Business Logic overview

The Revenue Sharing Model is a way to relay transactions so that the resulting relay fees are shared among multiple partners. 

A new type of contract (called "Collector") can now be used in relay requests, and will act as the receiver of its related relay fees. The usual receivers of these fees are the relay workers, and this will continue to be the case if a Collector is not used. This is, in fact, the way to opt out of the Revenue Sharing feature. 

At a later (and arbitrary) point in time, collected fees can be withdrawn from the contract and distributed between the different partners involved (also called "beneficiaries") which are addresses written in the contract, each with their own percentage of shares.  

For more in-depth information please see [the devportal Revenue Sharing documentation](https://github.com/mortelli/devportal/pull/20). 

## Technical overview
1. Config parameters for the collector contract and partners has been added so that their balances are shown.
2. Fees are now optional from the javascript side (applies to the `amount` field).
4. The README file has been improved.
5. React & front-end fixes/improvements have been applied.
6. The `package-lock` file has been regenerated.

### Other Revenue Sharing PRs
1. `rif-relay-contracts`: [PR #22 (Revenue Sharing Model)](https://github.com/anarancio/rif-relay-contracts/pull/22)
7. `rif-relay-common`: [PR #10 (Revenue Sharing Model)]( https://github.com/infuy/rif-relay-common/pull/10)
8. `rif-relay-client`: [PR #12 (Revenue Sharing Model)](https://github.com/infuy/rif-relay-client/pull/12)
9. `rif-relay-server`: [PR #16 (Revenue Sharing Model)](https://github.com/infuy/rif-relay-server/pull/16)
10. `rif-relay`: [PP #194 (Revenue Sharing Model)](https://github.com/rsksmart/rif-relay/pull/194)
11. `relaying-servives-sdk`: [PR #14 (Revenue Sharing Model)](https://github.com/infuy/relaying-services-sdk/pull/14) 
12. `devportal`: [Revenue Sharing documentation](https://github.com/mortelli/devportal/pull/20)

### Revenue Sharing Jira issues
See [here](https://rsklabs.atlassian.net/issues/?jql=labels%20%3D%20%22revenue-sharing%22).

### Other 
[RIF Relay - Revenue Sharing implementation](https://docs.google.com/document/d/1lmh78Bv7my5z6gkLEHuSqAj4NIQxZw6JwwfFT6zd9Dw)

### CODE REVIEW

**Comment by mortelli** [comment]
will update with technical details/review as soon as the merge conflicts are solved

**Comment by antomor** [comment]
Created a [new PR](https://github.com/antomor/relaying-services-sdk-dapp/pull/14) to solve the conflicts.

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "react-scripts": "4.0.3",
-    "relaying-services-sdk": "github:infuy/relaying-services-sdk",
```
Comment (mortelli) [inline package.json]: must be updated when the SDK PR is merged

**Review by antomor — APPROVED** [review]
LGTM!
Once we tackle all the open issues on the other PRs, I'd like to test the dapp before merging

**Pull request merged**